rust/lance/src/dataset/fragment/write.rs

```diff
@@ -23,6 +23,54 @@ use crate::dataset::write::do_write_fragments;
 use crate::dataset::{WriteMode, WriteParams, DATA_DIR};
 use crate::Result;
 
+/// Generates a filename optimized for S3 throughput using a UUID-based approach.
+///
+/// This approach follows Apache Iceberg's ObjectStoreLocationProvider pattern:
+/// - Takes a UUID (16 bytes total)
+/// - Uses first 3 bytes (24 bits) as binary string prefix for S3 distribution
+/// - Uses remaining 13 bytes as hex string for uniqueness
+///
+/// Format: `<24-bit-binary><remaining-hex>`
+/// Example: "101100101101010011010110a1b2c3d4e5f6g7h8i9j0"
```

Comment:
Note that here we don't do the additional slashes (`1001/1001/1001/10011001`) like Iceberg, because that is only going to harm the performance (although not a lot). The main reason for doing that is to be friendly to HDFS file listing because it does not paginate, but that is probably the platform we care the least about, so I just removed the slashes.